src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            last = newNode;
+            first = newNode;
+        } else {
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        Node<T> newNode;
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkRange(index);
+        if (index == 0) {
+            newNode = new Node<>(null, value, first);
+            newNode.next = first;
+            first.prev = newNode;
+            first = newNode;
+        } else {
+            addToMiddle(value,index);
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T elements : list) {
+            add(elements);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkRange(index);
+        Node<T> itemNode;
+        if (size / 2 <= index) {
+            itemNode = getNodeByIndexFromLast(index);
+            return itemNode.item;
+        } else {
+            itemNode = getNodeByIndexFromFirst(index);
+        }
+        return itemNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkRange(index);
+        Node<T> current = getNodeByIndexFromFirst(index);
+        T oldValue = current.item;
+        current.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkRange(index);
+        T deletedItem;
+        Node<T> newNode;
+        if (index == 0) {
+            newNode = first;
+            deletedItem = newNode.item;
+            first = first.next;
+            size--;
+            return deletedItem;
+        }
+        if (index == size - 1) {
```

Comment:
Probably better with else if.
```suggestion
        } else if (index == size - 1) {
```